### PR TITLE
add new test models for Resize layer

### PR DESCRIPTION
**Pull request by JulienMaille**
https://github.com/opencv/opencv/issues/18695

### CODE REVIEW

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from __future__ import print_function
 import torch
-from torch.autograd import Variable
+from torch.autograd import Variable, Function
```
Comment (l-bat) [inline testdata/dnn/onnx/generate_onnx_models.py:3]: Why import `Function` is needed?

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 2]
```diff
@@ -652,7 +652,28 @@ def forward(self, x, y):
...
-save_data_and_model_multy_inputs("dynamic_resize", model, input_0, input_1, version=11)
+save_data_and_model_multy_inputs("dynamic_resize_9", model, input_0, input_1, version=9)
+save_data_and_model_multy_inputs("dynamic_resize_10", model, input_0, input_1, version=10)
+save_data_and_model_multy_inputs("dynamic_resize_11", model, input_0, input_1, version=11)
+save_data_and_model_multy_inputs("dynamic_resize_12", model, input_0, input_1, version=12)
+
+class DynamicResizeScale(nn.Module):
+    def __init__(self):
+        super(DynamicResizeScale, self).__init__()
+
```
Comment (l-bat) [inline testdata/dnn/onnx/generate_onnx_models.py:663]: Do we really need a constructor for this model?

File: testdata/dnn/onnx/generate_onnx_models.py [hunk 3]
```diff
@@ -652,7 +652,28 @@ def forward(self, x, y):
...
+    def forward(self, x, y):
+        h = y.size(2)
+        w = y.size(3)
```
Comment (l-bat) [inline testdata/dnn/onnx/generate_onnx_models.py:666]: Why these variables is needed?

**Comment by l-bat** [comment]
Please change "base" branch of this PR: master => 3.4  as mentioned in https://github.com/opencv/opencv/pull/18708#issuecomment-728366298

File: testdata/dnn/onnx/generate_onnx_models.py (hunk 3 above)
Comment (JulienMaille) [reply testdata/dnn/onnx/generate_onnx_models.py:666]: Yes, correct, i'll remove them

File: testdata/dnn/onnx/generate_onnx_models.py (hunk 1 above)
Comment (JulienMaille) [reply testdata/dnn/onnx/generate_onnx_models.py:3]: For this class:
https://github.com/opencv/opencv_extra/blob/052200ca49780a922cac144d69852021897a47d2/testdata/dnn/onnx/generate_onnx_models.py#L609-L616

File: testdata/dnn/onnx/generate_onnx_models.py (hunk 2 above)
Comment (JulienMaille) [reply testdata/dnn/onnx/generate_onnx_models.py:663]: Well 100% of all models are defined with a constructor.

Comment (l-bat) [reply testdata/dnn/onnx/generate_onnx_models.py:663]: Why not?
```python3
class DynamicResize(nn.Module):
    def forward(self, x, y):
        up = nn.Upsample(scale_factor=(0.5, 0.5), mode='bilinear')
        return up(x) + y
```

Comment (JulienMaille) [reply testdata/dnn/onnx/generate_onnx_models.py]: done

**Pull request merged**